src/storage.rs

```diff
@@ -41,17 +43,72 @@ where
         path: &str,
         skip: i64,
         page_size: i64,
-    ) -> Result<DataCollection, StorageError> {
+    ) -> Result<DataCollection, DataStorerError> {
         self.deref().get_collection(path, skip, page_size).await
     }
 
-    async fn create(&self, value: Data) -> Result<bool, StorageError> {
+    async fn create(&self, value: Data) -> Result<bool, DataStorerError> {
         self.deref().create(value).await
     }
 }
 
+/// Stores an instance of a redact-backed data storer, including a cache.
+#[derive(Clone)]
+pub struct CachedDataStorer<T: DataStorer, V: DataCacher> {
+    storer: T,
+    cacher: V
+}
+
+impl<T: DataStorer, V: DataCacher> CachedDataStorer<T, V> {
+    /// Instantiates a cached redact-backed data storer using an existing storer and cacher.
+    pub fn new(storer: T, cacher: V) -> CachedDataStorer<T,V>
+        where
+            T: DataStorer,
+            V: DataCacher
+    {
+        CachedDataStorer {
+            storer,
+            cacher,
+        }
+    }
+}
+
+#[async_trait]
+impl<T: DataStorer, V: DataCacher> DataStorer for CachedDataStorer<T, V> {
+    async fn get(&self, path: &str) -> Result<Data, DataStorerError> {
+        let cache_hit = self.cacher.exists(path).await?;
+        if cache_hit {
```

Comment:
double check if redis will return a not found if the data has expired

Reply:
per redis EXPIRE command docs:

> Set a timeout on key. After the timeout has expired, the key will automatically be deleted.